### PR TITLE
cache storage id mapping in memcache

### PR DESCRIPTION
**Pull request by icewind1991**
Saves a query for every storage that is being used

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

**Review by CarlSchwan — APPROVED** [review]

File: lib/private/Files/Cache/StorageGlobal.php [hunk 1]
```diff
@@ -95,24 +92,56 @@ public function getStorageInfo(string $storageId): ?array {
...
+	}
+
+	public function removeFromMemcache(int $numericId, string $id) {
+		$this->memCache->remove("id::$id");
+		$this->memCache->remove("numeric::$numericId");
+	}
+
 	public function clearCache() {
 		$this->cache = [];
+		$this->memCache->clear();
```
Comment (juliushaertl) [inline lib/private/Files/Cache/StorageGlobal.php:145]: Might be something to check where this is being called, clearing redis with a prefix might be a bit slow due to the keys command having a O(n) runtime.

**Comment by szaimen** [comment]
Apparently tests need fixing...

**Comment by skjnldsv** [comment]
```
1) OCA\Files_External\Tests\Storage\WebdavTest::testCheckUpdate
Trying to access array offset on value of type null

/home/runner/work/server/server/lib/private/Files/Cache/StorageGlobal.php:84
/home/runner/work/server/server/lib/private/Files/Cache/Storage.php:102
/home/runner/work/server/server/lib/private/Files/Cache/Storage.php:81

```

**Pull request closed without merging**